api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDao.java

```diff
@@ -652,12 +652,16 @@ protected void handleIdentifier(Criteria criteria, TokenAndListParam identifier)
 	}
 	
 	protected void handleNames(Criteria criteria, StringAndListParam name, StringAndListParam given,
-	        StringAndListParam family) {
+	        StringAndListParam family, Boolean isInnerQuery) {
 		if (name == null && given == null && family == null) {
 			return;
 		}
 		
-		criteria.createAlias("names", "pn");
+		if (isInnerQuery) {
+			criteria.createAlias("m.names", "pn");
+		} else {
+			criteria.createAlias("names", "pn");
+		}
```

Comment:
Instead of doing this, I think it would be better to do something like this:

```java
protected void handleNames(Criteria criteria, StringAndListParam name, StringAndListParam given, StringAndListParam family) {
    return handleNames(criteria, name, given, family, null);
}

protected void handleNames(Criteria criteria, StringAndListParam name, StringAndListParam given, StringAndListParam family, String personAlias) {
    if (name == null & given == null && family == null) {
        return;
    }

    if (StringUtils.isNotBlank(personAlias) {
        criteria.createAlias(String.format("%.names", personAlias), "pn");
    } else {
        criteria.createAlias("names", "pn");
    }
    
    ...
}
```

This doesn't break existing code and, to me, at any rate `personAlias` tells me what that parameter is and why it's important more than `isInnerQuery` does.

Reply:
of course, it's a great way. Sorry for this.